test/error-handling-headers.test.js

```diff
@@ -0,0 +1,59 @@
+const t = require('tap')
```

Comment:
```suggestion
'use strict'

const t = require('tap')
```

Do we need a new file? Where is this code already tested? Can this test not live there?

Reply:
why not `reply.test.js`?

Reply:
Just moved the tests to reply-error.test.js instead.